### PR TITLE
(7.1.0 branch) mandatory and default parameters for sequence template

### PR DESCRIPTION
**Pull request by nilminiwso2**
Fixes https://github.com/wso2/docs-ei/issues/1701
Document mandatory and default parameters for sequence template

### CODE REVIEW

File: en/micro-integrator/docs/references/mediators/call-Template-Mediator.md [hunk 1]
```diff
@@ -145,3 +152,39 @@ dynamically set via the message context.
...
 ```
+
+### Example 3
+
+Consider an example where the sequence template is configured to log the greeting message that is passed from the mediation sequence in the REST API. According to the sequence template, a value for the greeting message is mandatory. 
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<template name="sequence-temp" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter isMandatory="false" defaultValue="Welcome" name="greeting_message"/>
```
Comment (abeykoon) [inline en/micro-integrator/docs/references/mediators/call-Template-Mediator.md:163]: `a value for the greeting message is mandatory` but here isMandatory="false". Seems there is a conflict.

File: en/micro-integrator/docs/references/mediators/call-Template-Mediator.md [hunk 2]
```diff
@@ -145,3 +152,39 @@ dynamically set via the message context.
...
+    <sequence>
+        <log level="custom">
+            <property expression="$func:greeting_message" name="greeting"/>
+        </log>
+    </sequence>
+</template>
+```
+
+However, this REST API is not passing a greeting message to this template. Also, a <b>default</b> greeting message is not defined in the template. To handle this error <b>at the point of calling the template</b>, you can add the 'onError' parameter to the Call Template mediator and call an error-handling sequence.
```
Comment (abeykoon) [inline en/micro-integrator/docs/references/mediators/call-Template-Mediator.md:172]: However, in the following example REST API is not passing a greeting message to this template. Also, a <b>default</b> greeting message is not defined in the template. In that case an error will get triggered when calling the template. If you need to handle this error, or in general any error that may occur at execution of the mediation logic inside the template, you can add the...

File: en/micro-integrator/docs/references/synapse-properties/template-properties.md [hunk 3]
```diff
@@ -31,14 +31,75 @@ See the topics given below for the list of properties that can be configured whe
...
+        </td>
+    </tr>
+    <tr>
+        <td>
+            isMandatory
+        </td>
+        <td>
+            Set this parameter to 'true' if a value is required to be passed for the configuration.</br></br>
+            When this parameter is 'true', if a value is not passed from a Call Template mediator as shown below, the Micro Integrator will first check for a <a href="#defaultValue">defaultvalue</a>. If a <a href="#defaultValue">defaultValue</a> does not exist, the 'onError' sequence of the <a href="../../references/synapse-properties/template-properties/call-Template-Mediator">Call Template mediator</a> will be called.</br></br>
+            If an empty value is passed in the Call Template mediator, the empty value will be used instead of using the default value or the onError sequence.</br></br>
```
Comment (abeykoon) [inline en/micro-integrator/docs/references/synapse-properties/template-properties.md:90]: or the onError sequence. - this is not needed.

File: en/micro-integrator/docs/references/mediators/call-Template-Mediator.md (hunk 1 above)
Comment (nilminiwso2) [reply en/micro-integrator/docs/references/mediators/call-Template-Mediator.md]: Fixed

**Review by RukshiW — APPROVED** [review]

**Pull request merged**